src/NuGet.Indexing/Extraction/CatalogPackageMetadataExtraction.cs

```diff
@@ -124,6 +125,42 @@ private void AddListed()
                 _metadata["listed"] = listed;
             }
 
+            private void AddSemVerLevelKey()
+            {
+                var version = (string)_catalog["verbatimVersion"];
+                if (version != null)
+                {
+                    NuGetVersion packageOriginalVersion;
+                    if (NuGetVersion.TryParse(version, out packageOriginalVersion))
+                    {
+                        if (packageOriginalVersion.IsSemVer2)
+                        {
+                            _metadata["semVerLevelKey"] = "2";
+                            return;
+                        }
+                    }
+                }
+
+                var dependencyGroups = _reader.GetPackageDependencies().ToList();
+                foreach (var dependencyGroup in dependencyGroups)
+                {
+                    if (dependencyGroup.Packages.Any())
+                    {
+                        // Add packages list
+                        foreach (var packageDependency in dependencyGroup.Packages)
+                        {
+                            var versionRange = packageDependency.VersionRange;
+                            if ((versionRange.MaxVersion != null && versionRange.MaxVersion.IsSemVer2)
+                                || (versionRange.MinVersion != null && versionRange.MinVersion.IsSemVer2))
+                            {
+                                _metadata["semVerLevelKey"] = "2";
```

Comment:
The string `"semVerLevelKey"` is used in multiple places in this PR: extract const. Same for the value `"2"`.